src/client.js

```diff
@@ -0,0 +1,115 @@
+import webapi, { LogLevel } from "@slack/web-api";
+import { HttpsProxyAgent } from "https-proxy-agent";
+import Config from "./config.js";
+import SlackError from "./errors.js";
+
+/**
+ * The Client class creates a WebClient from @slack/web-api for use when calling
+ * various Slack API methods.
+ *
+ * @see {@link https://slack.dev/node-slack-sdk/web-api/}
+ * @see {@link https://api.slack.com/methods/}
+ */
+export default class Client {
+  /**
+   * Possible response values related to messages from the API.
+   * @typedef MessageResult - Possible message values from API methods.
+   * @prop {string} [threadTs] - timestamp of the top threaded message.
+   * @prop {string} [ts] - timestamp of the message.
+   * @prop {string} [channel] - ID of the channel.
+   * @see {@link https://api.slack.com/methods/chat.postMessage#examples}
+   */
+
+  /**
+   * Perform the API call configured with the input payload.
+   * @param {Config} config
+   */
+  async post(config) {
+    if (!config.inputs.method) {
+      throw new SlackError(config.core, "No API method was provided for use");
+    }
+    if (!config.inputs.token) {
+      throw new SlackError(config.core, "No token was provided to post with");
+    }
+    const client = new config.webapi.WebClient(config.inputs.token, {
+      agent: this.proxies(config)?.httpsAgent,
+      retryConfig: this.retries(config.inputs.retries),
+      logger: {
+        debug: config.core.debug,
+        info: config.core.info,
+        warn: config.core.warning,
+        error: config.core.error,
+        getLevel: () => {
+          return config.core.isDebug() ? LogLevel.DEBUG : LogLevel.INFO;
+        },
+        setLevel: (_level) => {},
+        setName: (_name) => {},
+      },
```

Comment:
Right now we don't have an option to adjust the log level which seems alright for now IMO. For reference, this is [what a warning looks like](https://github.com/zimeg/slack-github-action/actions/runs/11042479420) and [here are some debug logs](https://github.com/zimeg/slack-github-action/actions/runs/11042479420/job/30675170468#step:11:78)! 🔍